### ISSUE TITLE
[Firefox Containers] Can't login into account if Phanpy or account opened inside container

### ISSUE THREAD

**Issue by SadmL** [cheeaun/phanpy#667]
**Describe the bug**
- Self-hosted Phanpy (bare-metal, serving dist folder) [phanpy.sadium.cyou](https://phanpy.sadium.cyou)
- Version 2024.08.22. d54cb61
- I've tested Mastodon [social.exo.icu](https://social.exo.icu) and self-hosted GoToSocial [gts.sadium.cyou](https://gts.sadium.cyou)

**To Reproduce**
1. Go to [phanpy.sadium.cyou](https://phanpy.sadium.cyou) (containerized or not)
2. Click on 'Login'
3. Authorize in instance (containerized or not, opposite to step 1, or both containerized)
4. Looks like login happens, but no account was added into Phanpy

**Expected behavior**
Go to [phanpy.sadium.cyou](https://phanpy.sadium.cyou) -> Login -> Authorize -> Account is added in Phanpy (repeat x2 in my case)

**Screenshots**
No login
![изображение](https://github.com/user-attachments/assets/90a1cbd8-e585-411d-bcb2-3c93504ae50e)

Logged in
![изображение](https://github.com/user-attachments/assets/374fd22d-abe4-485d-a63d-9f807daf5d32)

**Desktop (please complete the following information):**
 - OS: EndeavourOS (Arch-based) | KDE Plasma 6.1.4
 - Browser: Librewolf
 - Version: 129.0.1-1

**Smartphone (please complete the following information):**
 - Device: X
 - OS: X
 - Browser X
 - Version X

**Additional context**
If no containers used or account and Phanpy inside same container, no problems with login. [Elk](https://github.com/elk-zone/elk) client doesn't have such problem.
#285 may be related, but both instances uses subdomains as @subdomain.domain.com

**Comment by cheeaun** [comment]
If login works without containers, it's probably not a subdomain issue.

This is a bit hard for me to replicate. Are you able to debug this on your side, maybe looking at the browser devtools console? 🙏

**Comment by SadmL** [comment]
If I know how to debug 😅
May be this will be helpful
![изображение](https://github.com/user-attachments/assets/e90a9993-2890-4f23-bfa1-7369182f8fe5)

**Comment by SadmL** [comment]
I don't think that is OS related, more likely just Firefox containers. For example I want to use Github for Crowdin login.
Github in container 1
Crowdin - no container
No container Crowdin - Press login - redirect to containerized Github - Github authorization - redirects back into Crowdin with no container and Github based account

**Comment by cheeaun** [comment]
@SadmL if this is your own server and the admin, is it possible if you create a temporary dummy account for me to test? [Private mention](https://mastodon.social/@cheeaun) me the details.

**Comment by SadmL** [comment]
Check DM

**Comment by SadmL** [comment]
I've rebuild with latest https://github.com/cheeaun/phanpy/commit/58ccdf84fad03f24e04f815e6fed331813043f12 and it's looks like working now

**Comment by anhgelus** [comment]
If you are switching containers during the login (e.g., phanpy is in container A and your instance is in container B), it sends you an HTTP 401 or an HTTP 403
